### PR TITLE
Initial implementation of NJsonSchema as replacement for Newtonsoft Json Schema 

### PR DESCRIPTION
**Pull request by WilliamBerryiii**

### CODE REVIEW

**Comment by WilliamBerryiii** [comment]
@suneetnangia - this is an initial spike of the NJsonSchema implementation for review.

**Comment by WilliamBerryiii** [comment]
I'll get the json-everything spike rolling now.

File: common/src/Microsoft.Azure.IIoT.Serializers.NewtonSoft/src/Default/NewtonSoftJsonSerializer.cs [hunk 1]
```diff
@@ -72,21 +74,24 @@ public class NewtonSoftJsonSerializer : IJsonSerializerSettingsProvider,
...
-                                validatingReader.ValidationEventHandler += (o, a) => messages.Add(a.Message);
-                                var serializedType = jsonSerializer.Deserialize(validatingReader, type);
-
-                                return messages.Count > 0 ? throw new JsonSerializationException(string.Join(", ", messages.ToArray())) : serializedType;
-                            }
+                        // the inbound data 
+                        using (var validationDataStream = new MemoryStream(bufferArray))
+                        using (var validationReader = new StreamReader(validationDataStream, ContentEncoding)) {
+                            
+                            var schema = Task.Run(() => JsonSchema.FromJsonAsync(schemaReader.ReadToEnd())).GetAwaiter().GetResult();
```
Comment (suneetnangia) [inline common/src/Microsoft.Azure.IIoT.Serializers.NewtonSoft/src/Default/NewtonSoftJsonSerializer.cs:87]: Wonder if we need Task.Run() here, could we simplify it i.e. "JsonSchema.FromJsonAsync(schemaReader.ReadToEnd()).GetAwaiter().GetResult();"?

Comment (WilliamBerryiii) [reply common/src/Microsoft.Azure.IIoT.Serializers.NewtonSoft/src/Default/NewtonSoftJsonSerializer.cs:87]: As an FYI: https://stackoverflow.com/a/59019473/1276028

**Comment by WilliamBerryiii** [comment]
@suneetnangia - I'm going to nuke this PR and the associated branch as we've committed to the json-everything lib for schema management. Cool?

**Pull request closed without merging**